src/utilities/IOManager.cpp

```diff
@@ -10,128 +10,95 @@
 
 namespace rtt::ai::io {
 
-std::mutex IOManager::worldStateMutex;
-std::mutex IOManager::geometryMutex;
-std::mutex IOManager::robotFeedbackMutex;
-std::mutex IOManager::refereeMutex;
-std::mutex IOManager::demoMutex;
 
 IOManager io;
 
 IOManager::~IOManager() {
     delete worldSubscriber;
-    delete geometrySubscriber;
-    delete refSubscriber;
-    delete feedbackSubscriber;
     delete robotCommandPublisher;
     delete settingsPublisher;
 }
 
 void IOManager::init(int teamId) {
     RTT_INFO("Setting up IO publishers/subscribers")
-    worldSubscriber = new proto::Subscriber<proto::World>(proto::WORLD_CHANNEL, &IOManager::handleWorldState, this);
-    geometrySubscriber = new proto::Subscriber<proto::SSL_GeometryData>(proto::GEOMETRY_CHANNEL, &IOManager::handleGeometry, this);
-    refSubscriber = new proto::Subscriber<proto::SSL_Referee>(proto::REFEREE_CHANNEL, &IOManager::handleReferee, this);
+    worldSubscriber = new proto::Subscriber<proto::State>(proto::WORLD_CHANNEL, &IOManager::handleState, this);
 
     // set up advertisement to publish robotcommands and settings
     if (teamId == 1) {
-        feedbackSubscriber = new proto::Subscriber<proto::RobotFeedback>(proto::FEEDBACK_SECONDARY_CHANNEL, &IOManager::handleFeedback, this);
-        robotCommandPublisher = new proto::Publisher<proto::RobotCommand>(proto::ROBOT_COMMANDS_SECONDARY_CHANNEL);
+        robotCommandPublisher = new proto::Publisher<proto::AICommand>(proto::ROBOT_COMMANDS_SECONDARY_CHANNEL);
         settingsPublisher = new proto::Publisher<proto::Setting>(proto::SETTINGS_SECONDARY_CHANNEL);
     } else {
-        feedbackSubscriber = new proto::Subscriber<proto::RobotFeedback>(proto::FEEDBACK_PRIMARY_CHANNEL, &IOManager::handleFeedback, this);
-        robotCommandPublisher = new proto::Publisher<proto::RobotCommand>(proto::ROBOT_COMMANDS_PRIMARY_CHANNEL);
+        robotCommandPublisher = new proto::Publisher<proto::AICommand>(proto::ROBOT_COMMANDS_PRIMARY_CHANNEL);
         settingsPublisher = new proto::Publisher<proto::Setting>(proto::SETTINGS_PRIMARY_CHANNEL);
     }
+    central_server_connection = new networking::PairReceiver<16970>();
```

Comment:
no.

Reply:
Doesn't need to be heap allocated in general.

Reply:
I don't like this, because now the connection is started at an undetermined time when the application is started. I'd rather have a heap allocation so it's easier in the future to close and restart the connection rather than this model.

Reply:
It makes it easier to see when and how the connection is started.

Reply:
If it was only a constructor, that's fine, but this also starts the connection, so that's why I disagree.

Reply:
Sure, fair enough.